### PR TITLE
legend refactor

### PR DESCRIPTION
**Pull request by dariuszlacheta**

### CODE REVIEW

File: packages/charts/src/components/legend/series/series-horizontal.component.tsx [hunk 1]
```diff
@@ -1,56 +1,188 @@
...
+              </Layout>
+            ) : (
+              <Slider
+                mode="horizontal"
+                dimension={sliderDimension}
+                nextDisabled={endOffset === dataSeries.length}
+                previousDisabled={startOffset === 0}
+                direction={direction}
+                animation={(itemIndex) => {
+                  const itemPosition =
```
Comment (maciejrybaniec) [inline packages/charts/src/components/legend/series/series-horizontal.component.tsx:135]: As we discussed - the animation logic become too complex so please, extract it to separate function responsible for creation of transition.

**Review by maciejrybaniec — CHANGES_REQUESTED** [review]

File: packages/charts/src/components/legend/series/series-horizontal.component.tsx (hunk 1 above)
Comment (dariuszlacheta) [reply packages/charts/src/components/legend/series/series-horizontal.component.tsx]: Good call 👍  done

File: packages/charts/src/components/legend/series/series.component.tsx [hunk 2]
```diff
@@ -28,57 +28,79 @@ type Props = {
...
+  );
+
   return (
     <>
       {layout === 'horizontal' ? (
-        <SeriesHorizontal {...commonProps}>{items}</SeriesHorizontal>
+        <SeriesHorizontal
+          {...commonProps}
+          renderNodes={renderNodes}
+          onOffsetUpdate={(offset: [number, number]) => {
```
Comment (maciejrybaniec) [inline packages/charts/src/components/legend/series/series.component.tsx:92]: For initial release - please make `onOffsetUpdate` optional and remove properties from SeriesHorizontal and SeriesVertical components.

**Review by maciejrybaniec — CHANGES_REQUESTED** [review]
Looks good for me. Just small change is needed to clear out not required code.

File: packages/charts/src/components/legend/series/series.component.tsx (hunk 2 above)
Comment (dariuszlacheta) [reply packages/charts/src/components/legend/series/series.component.tsx]: Done

**Review by maciejrybaniec — APPROVED** [review]

**Pull request merged**